packages/contracts-bridge/contracts/BridgeMessage.sol

```diff
@@ -130,22 +131,32 @@ library BridgeMessage {
         return isType(_action, Types.FastTransfer);
     }
 
+    /**
+     * @notice Checks that the message is of type ConnextTransfer
+     * @param _action The message
+     * @return True if the message is of type ConnextTransfer
+     */
+    function isConnextTransfer(bytes29 _action) internal pure returns (bool) {
+        return isType(_action, Types.ConnextTransfer);
+    }
+
+
     /**
      * @notice Formats Transfer
      * @param _to The recipient address as bytes32
      * @param _amnt The transfer amount
      * @param _detailsHash The hash of the token name, symbol, and decimals
+     * @param _externalId The external identifier of the transfer
      * @return
      */
     function formatTransfer(
         bytes32 _to,
         uint256 _amnt,
-        bytes32 _detailsHash
+        bytes32 _detailsHash,
+        bytes32 _externalId
     ) internal pure returns (bytes29) {
-        return
-            abi.encodePacked(Types.Transfer, _to, _amnt, _detailsHash).ref(0).castTo(
-                uint40(Types.Transfer)
-            );
+        Types transferType = _externalId == bytes32(0) ? Types.Transfer : Types.ConnextTransfer;
```

Comment:
we should not change the content/length/format of `Types.Transfer` - the new Type should not modify the old types - for backwards compatibility